### ISSUE TITLE
[bug] `logprob_chunk_size` is broken in distillation and can cause OOM

### ISSUE THREAD

**Issue by odedovadia** [NVIDIA-NeMo/RL#2882]
**Describe the bug**

`logprob_chunk_size` does not bound peak GPU memory in `distributed_vocab_topk`
and `gather_logits_at_global_indices` (`nemo_rl/distributed/model_utils.py`) in distillation.
Both functions upcast the entire `[B, S, V_local]` logits tensor to fp32
**before** the per-chunk loop, so the fp32 working set scales with the full
sequence length `S` instead of `chunk_size`. The chunking therefore gives no
memory benefit, and long sequences (e.g. distillation at 32K) hit avoidable OOMs.

The offending line (hoisted above the loop) in each function:
- [`distributed_vocab_topk`](https://github.com/NVIDIA-NeMo/RL/blob/29fc948c5debe13373616b56661ab164696dec6c/nemo_rl/distributed/model_utils.py#L1452)
- [`gather_logits_at_global_indices`](https://github.com/NVIDIA-NeMo/RL/blob/29fc948c5debe13373616b56661ab164696dec6c/nemo_rl/distributed/model_utils.py#L1541)

```python
logits = vocab_parallel_logits.to(dtype=torch.float32)  # whole [B, S, V_local] in fp32, before chunking
```

**Steps/Code to reproduce bug**

```python
import torch
from nemo_rl.distributed.model_utils import gather_logits_at_global_indices

B, S, V_local, k = 1, 32768, 16032, 64
logits = torch.randn(B, S, V_local, device="cuda", dtype=torch.bfloat16)
idx = torch.randint(0, V_local, (B, S, k), device="cuda")

for chunk in (None, 512):
    torch.cuda.reset_peak_memory_stats()
    gather_logits_at_global_indices(
        logits, idx, tp_group=None,
        vocab_start_index=0, vocab_end_index=V_local, chunk_size=chunk,
    )
    print(chunk, f"{torch.cuda.max_memory_allocated() / 1e6:.0f} MB")
# Expected: chunk_size=512 memory should be lower than chunk_size=None.
# Actual (B200, bf16): None -> 3213 MB, 512 -> 3186 MB (chunk_size ignored).
```

**Expected behavior**

Peak fp32 memory should scale with `chunk_size`, not the full sequence length,
so that `logprob_chunk_size` can bound memory for long-sequence distillation.

**Additional context**

**Comment by odedovadia** [comment]
Opened a PR #2883

**Comment by RayenTian** [comment]
Thanks for open this issue @odedovadia! The PR is under review.
